packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -33,63 +32,48 @@ const createEmptyResponse = (code: string, headers: IHttpHeaderParam[], mediaTyp
     O.map(() => ({ code, headers }))
   );
 
+type BodyNegotiationResult = Omit<IHttpNegotiationResult, 'headers'>;
+
 const helpers = {
   negotiateByPartialOptionsAndHttpContent(
     { code, exampleKey, dynamic }: NegotiatePartialOptions,
     httpContent: IMediaTypeContent
-  ): E.Either<Error, Omit<IHttpNegotiationResult, 'headers'>> {
+  ): E.Either<Error, BodyNegotiationResult> {
     const { mediaType } = httpContent;
 
     if (exampleKey) {
-      // the user provided exampleKey - highest priority
-      const example = findExampleByKey(httpContent, exampleKey);
-      if (example) {
-        // example exists, return
-        return E.right({
-          code,
-          mediaType,
-          bodyExample: example,
-        });
-      } else {
-        return E.left(
+      return pipe(
+        findExampleByKey(httpContent, exampleKey),
+        E.fromOption(() =>
           ProblemJsonError.fromTemplate(
             NOT_FOUND,
             `Response for contentType: ${mediaType} and exampleKey: ${exampleKey} does not exist.`
           )
-        );
-      }
+        ),
+        E.map(bodyExample => ({ code, mediaType, bodyExample }))
+      );
     } else if (dynamic === true) {
-      if (httpContent.schema) {
-        return E.right({
-          code,
-          mediaType,
-          schema: httpContent.schema,
-        });
-      } else {
-        return E.left(new Error(`Tried to force a dynamic response for: ${mediaType} but schema is not defined.`));
-      }
+      return pipe(
+        httpContent.schema,
+        E.fromNullable(new Error(`Tried to force a dynamic response for: ${mediaType} but schema is not defined.`)),
+        E.map(schema => ({ code, mediaType, schema }))
+      );
     } else {
       // try to find a static example first
-      const example = findBestExample(httpContent);
-      if (example) {
-        // if example exists, return
-        return E.right({
-          code,
-          mediaType,
-          bodyExample: example,
-        });
-      } else if (httpContent.schema) {
-        return E.right({
-          code,
-          mediaType,
-          schema: httpContent.schema,
-        });
-      } else {
-        return E.right({
-          code,
-          mediaType,
-        });
-      }
+
+      return E.right(
+        pipe(
+          findFirstExample(httpContent),
+          O.map(bodyExample => ({ code, mediaType, bodyExample })),
+          O.alt(() =>
+            pipe(
+              O.fromNullable(httpContent.schema),
+              O.map<JSONSchema, BodyNegotiationResult>(schema => ({ schema, code, mediaType }))
+            )
+          ),
+          O.getOrElse<BodyNegotiationResult>(() => ({ code, mediaType }))
+        )
```

Comment:
Is it possible to put here:
`O.fold(() => E.right({ code, mediaType }), E.right)`

and remove

`return E.right(` ?

Reply:
It can, but it would require additional explicit typings because of the sad shape of our data:

<img width="564" alt="image" src="https://user-images.githubusercontent.com/1416224/99967601-b82b8380-2d5d-11eb-85a4-acdc3dc6cd81.png">

Reply:
Uh